src/controller/static_mates.controller.ts

```diff
@@ -1,72 +1,114 @@
-//         res.status(500).json({ message: error.message });
-//     }
-// }
-// export const createStatic_Mate = async (req, res) => {
-//     try {
-//         const single_static_mate = await Static_Mates.create(req.body);
-//         res.status(200).json(single_static_mate);
-//     } catch (error) {
-//         res.status(500).json({ message: error.message })
-//     }
-// };
-
+//Pull a SINGLE PERSON NEEDS TO BE FIXED
+export const findSingleStaticMate = async (req, res) => {
+    try {
+        console.log("Calling find all for Static Mates")
+        const allStaticMates:StaticMate[] = await StaticMate.findAll();
+        res.status(200).json({message: allStaticMates})
+    } catch (error) {
+        res.status(500).json({ message: error.message });
+    }
+}
 
-// ///Update
-// export const updateStatic_Mate = async (req, res) => {
-//     try {
-//         const { id } = req.params;
+///create
 
-//         const single_static_mate = await Static_Mates.findByIdAndUpdate(id, req.body);
+export const createSingleStaticMate = async (req, res) => {
+    try {
+        //properties on the query object and the query object is property on the request object 
+        const { static_name, raid_member_role } = req.body;
+        //console.log("req.body", req.body)
+        //find a single pesron
+        const sM = await StaticMate.findOne({ where: {name: static_name} });
+        if (sM){
+            throw Error("There was already a user with that name")
+            return
+        }
+        //this is the set up of paramaters for postman
+        const single_static_mate = await StaticMate.create({
+                name: static_name ,
+                role: raid_member_role,
+                createdAt: new Date(),
+                updatedAt: new Date(),
+        })
+        //this is confirming that things worked
+        if (single_static_mate){
+            console.log("user was created", single_static_mate)
+        }
+        else {
+            throw Error("something went wrong")
+        }
+        res.status(200).json({message: "static member was created"});
+    }
+    catch (error){
+        console.log("we had an error: ", error)
+        res.status(500).json({ message: error.message });
+    }
+}
 
-//         if (!single_static_mate) {
-//             return res.status(404).json({ message: "Product not found" });
-//         }
-//         const updated_static_mate = await Static_Mates.findById(id);
-//         res.status(200).json(updated_static_mate);
+;
 
-//     } catch (error) {
-//         res.status(500).json({ message: error.message });
-//     }
+/////Update
+export const updateStaticMate = async (req, res) => {
+    try {
+        //do a find first to see if that thing exist
+        //also ends things early
+        const {id} = req.params;
+        const {static_name, raid_member_role} = req.body;
+        console.log("req.body is reading", req.body)
+        const sM = await StaticMate.findOne({ where: {id} });
```

Comment:
.

Reply:
Update has been updated akin to the other controllers o7